eth2util/rlp/rlp_internal_test.go

```diff
@@ -0,0 +1,30 @@
+// Copyright © 2022-2023 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package rlp
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDecodeLength(t *testing.T) {
+	t.Run("ok", func(t *testing.T) {
+		items := []byte{0xb8, 0x01, 0x0a}
+		offset, length, err := decodeLength(items)
+		require.NoError(t, err)
+		require.Equal(t, 2, offset)
+		require.Equal(t, 1, length)
+	})
+
+	t.Run("prefix absent", func(t *testing.T) {
+		_, _, err := decodeLength(nil)
+		require.ErrorContains(t, err, "input too short")
+	})
+
+	t.Run("prefix provided but length absent", func(t *testing.T) {
```

Comment:
what about a test case for the code you changed: `invalid length prefix`?

Reply:
missed it, adding it now

Reply:
On further thought, i think the check `length > 8 || length <= 0` is not possible altogether, it should be a panic if it happens. It should therefore be removed IMO.

Reasoning: 
```
Since
0xb8 <= prefix < 0xc0
=> 184 <= prefix < 192 
=> 184-183 <= prefix-0xb7 < 192-183
=> 1 <= length < 9
=> length range [1,8]
```
If we provide a `value < 0xb8` to trigger `length == 0`, it will be caught before the code block by:
```go
    if prefix < 0xb8 {
        return 1, int(prefix - 0x80), nil
    }
```
And if we provide a `value >= 0xc0`, it will skip the code block altogether:
```go
if prefix < 0xc0 { ... }
```